crates/win-api-wrappers/src/service.rs

```diff
@@ -0,0 +1,150 @@
+use std::alloc::Layout;
+
+use windows::core::Owned;
+use windows::Win32::Foundation::{ERROR_INSUFFICIENT_BUFFER, GENERIC_READ};
+use windows::Win32::System::Services::{
+    OpenSCManagerW, OpenServiceW, QueryServiceConfigW, QueryServiceStatus, StartServiceW, QUERY_SERVICE_CONFIGW,
+    SC_HANDLE, SC_MANAGER_ALL_ACCESS, SERVICE_ALL_ACCESS, SERVICE_AUTO_START, SERVICE_BOOT_START, SERVICE_DEMAND_START,
+    SERVICE_DISABLED, SERVICE_QUERY_CONFIG, SERVICE_QUERY_STATUS, SERVICE_RUNNING, SERVICE_STATUS,
+    SERVICE_SYSTEM_START,
+};
+
+use crate::raw_buffer::RawBuffer;
+use crate::utils::WideString;
+
+pub struct ServiceManager {
+    handle: Owned<SC_HANDLE>,
+}
+
+impl ServiceManager {
+    pub fn open_read() -> anyhow::Result<Self> {
+        Self::open_with_access(GENERIC_READ.0)
+    }
+
+    pub fn open_all_access() -> anyhow::Result<Self> {
+        Self::open_with_access(SC_MANAGER_ALL_ACCESS)
+    }
+
+    fn open_with_access(access: u32) -> anyhow::Result<Self> {
+        // SAFETY: FFI call with no outstanding preconditions.
+        let raw_sc_handle = unsafe { OpenSCManagerW(None, None, access)? };
+
+        // SAFETY: Handle returned by `OpenSCManagerW` is valid and needs to be closed after use,
+        // thus it is safe to take ownership of it via `Owned`.
+        let handle = unsafe { Owned::new(raw_sc_handle) };
+
+        Ok(Self { handle })
+    }
+
+    fn open_service_with_access(&self, service_name: &str, access: u32) -> anyhow::Result<Service> {
+        let service_name_wide = WideString::from(service_name);
+
+        // SAFETY: Value passed as hSCManager is valid as long as `ServiceManager` instance is
+        // alive. Service name is valid null-terminated UTF-16 string allocated on the heap.
+        let raw_service_handle = unsafe { OpenServiceW(*self.handle, service_name_wide.as_pcwstr(), access)? };
```

Comment:
nitpick: In term on readability, I recommend a newline for each item, optionally formatted as a list.

nitpick: Use the name of the variable as-is.

suggestion: Two options:
```suggestion
        // SAFETY: 
        // - Value passed as hSCManager is valid as long as `ServiceManager` instance is alive.
        // - service_name_wide is a valid, null-terminated UTF-16 string allocated on the heap.
        let raw_service_handle = unsafe { OpenServiceW(*self.handle, service_name_wide.as_pcwstr(), access)? };
```
```suggestion
        // SAFETY: Value passed as hSCManager is valid as long as `ServiceManager` instance is alive. 
        // service_name_wide is a valid, null-terminated UTF-16 string allocated on the heap.
        let raw_service_handle = unsafe { OpenServiceW(*self.handle, service_name_wide.as_pcwstr(), access)? };
```